case_utils/case_validate/__init__.py

```diff
@@ -64,13 +65,175 @@ class NonExistentCDOConceptWarning(UserWarning):
     pass
 
 
+class ValidationResult:
+    conforms: bool
+    graph: Any
+    text: str
+    undefined_concepts: Set[rdflib.URIRef]
+
+
 def concept_is_cdo_concept(n_concept: rdflib.URIRef) -> bool:
+    """
+    Determine if a concept is part of the CDO ontology.
+
+    :param n_concept: The concept to check.
+    :return: whether the concept is part of the CDO ontology.
+    """
     concept_iri = str(n_concept)
     return concept_iri.startswith(
         "https://ontology.unifiedcyberontology.org/"
     ) or concept_iri.startswith("https://ontology.caseontology.org/")
 
 
+def get_ontology_graph(
+    case_version: Optional[str] = None, supplemental_graphs: Optional[List[str]] = None
+) -> rdflib.Graph:
+    """
+    Get the ontology graph for the given case_version and any supplemental graphs.
+
+    :param case_version: the version of the CASE ontology to use.  If None, the most recent version will be used.
+    :param supplemental_graphs: a list of supplemental graphs to use.  If None, no supplemental graphs will be used.
+    :return: the ontology graph against which to validate the data graph.
+    """
+    ontology_graph = rdflib.Graph()
+    if case_version and case_version != "none":
+        ttl_filename = case_version + ".ttl"
+        _logger.debug("ttl_filename = %r.", ttl_filename)
+        ttl_data = importlib.resources.read_text(case_utils.ontology, ttl_filename)
+        ontology_graph.parse(data=ttl_data, format="turtle")
+    if supplemental_graphs:
+        for arg_ontology_graph in supplemental_graphs:
+            _logger.debug("arg_ontology_graph = %r.", arg_ontology_graph)
+            ontology_graph.parse(arg_ontology_graph)
+    return ontology_graph
+
+
+def get_invalid_cdo_concepts(
+    data_graph: Graph, ontology_graph: Graph
+) -> Set[rdflib.URIRef]:
+    """
+    Get the set of concepts in the data graph that are not part of the CDO ontology.
+
+    :param data_graph: The data graph to validate.
+    :param ontology_graph: The ontology graph to use for validation.
+    :return: The list of concepts in the data graph that are not part of the CDO ontology.
```

Comment:
`s/ontology/ontologies/`, and `s/list/set/`, but otherwise the return line is good.

Also, it makes good sense to split this code hunk out into its own function.  If there are nontrivial review issues with this PR, I'll peel this small part out as its own PR.

Reply:
Fixed

Reply:
This change is exported to [PR 121](https://github.com/casework/CASE-Utilities-Python/pull/121) .

Reply:
This change has been integrated.